Patch from dtardon to fix iter advance in sallayout.cxx

From: Thorsten Behrens <tbehrens@novell.com>


---

 vcl/source/gdi/sallayout.cxx |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)


diff --git vcl/source/gdi/sallayout.cxx vcl/source/gdi/sallayout.cxx
index bd91f20..a3f4ff5 100755
--- vcl/source/gdi/sallayout.cxx
+++ vcl/source/gdi/sallayout.cxx
@@ -1804,8 +1804,8 @@ void MultiSalLayout::AdjustLayout( ImplLayoutArgs& rArgs )
     int nRunStart, nRunEnd;
     while (rArgs.GetNextRun(&nRunStart, &nRunEnd, &bRtl))
     {
-        if (bRtl) std::fill(vRtl.begin() + nRunStart - rArgs.mnMinCharPos,
-                            vRtl.begin() + nRunEnd - rArgs.mnMinCharPos, true);
+        if (bRtl) std::fill(vRtl.begin() + (nRunStart - rArgs.mnMinCharPos),
+                            vRtl.begin() + (nRunEnd - rArgs.mnMinCharPos), true);
     }
     rArgs.ResetPos();
 
